src/container/tabs.rs

```diff
@@ -108,60 +132,170 @@ impl Tabs {
-                        .drag_started()
-                    {
-                        ui.memory_mut(|mem| mem.set_dragged_id(tile_id.egui_id()));
-                    }
+            let mut scroll_state: ScrollState = ui
+                .ctx()
+                .memory_mut(|m| m.data.get_temp::<ScrollState>(id))
+                .unwrap();
+
+            const LEFT_FRAME_SIZE: f32 = 20.0;
+            const RIGHT_FRAME_SIZE: f32 = 20.0;
+
+            let mut consume = ui.available_width();
+
+            if (scroll_state.offset.x - RIGHT_FRAME_SIZE) > scroll_state.available.x {
+                if scroll_state.prev_frame_right {
+                    scroll_state.offset_delta.x += RIGHT_FRAME_SIZE;
                 }
 
-                for (i, &child_id) in self.children.iter().enumerate() {
-                    if !tree.is_visible(child_id) {
-                        continue;
-                    }
+                scroll_state.prev_frame_right = false;
+            } else if (scroll_state.offset.x - 0.0) > scroll_state.available.x {
+                // DO NOTHING
+            } else {
+                scroll_state.prev_frame_right = true;
+            }
+
+            if scroll_state.offset.x > LEFT_FRAME_SIZE {
+                if !scroll_state.prev_frame_left {
+                    scroll_state.offset_delta.x += LEFT_FRAME_SIZE;
+                }
 
-                    let is_being_dragged = is_being_dragged(ui.ctx(), child_id);
+                scroll_state.prev_frame_left = true;
 
-                    let selected = self.is_active(child_id);
-                    let id = child_id.egui_id();
+                consume -= LEFT_FRAME_SIZE;
+            } else if scroll_state.offset.x > 0.0 {
+                if scroll_state.prev_frame_left {
+                    scroll_state.offset.x -= LEFT_FRAME_SIZE;
+                }
 
-                    let response =
-                        behavior.tab_ui(&tree.tiles, ui, id, child_id, selected, is_being_dragged);
-                    let response = response.on_hover_cursor(egui::CursorIcon::Grab);
-                    if response.clicked() {
-                        next_active = Some(child_id);
-                    }
+                // Uncomment the following for an ~animated~ reveal.
+                // consume -= scroll_state.offset.x;
+            } else {
+                scroll_state.prev_frame_left = false;
+            }
+
+            if scroll_state.consumed.x > scroll_state.available.x
+                && (scroll_state.offset.x - RIGHT_FRAME_SIZE) < scroll_state.available.x
+            {
+                consume -= RIGHT_FRAME_SIZE;
+
+                behavior.top_bar_right_ui(
+                    &tree.tiles,
+                    ui,
+                    tile_id,
+                    self,
+                    scroll_state.offset.x,
+                    &mut scroll_state.offset_delta.x,
+                );
+            }
```

Comment:
I don't think this is good - the user should have the option to show buttons in the top-right of the tabs bar no matter what the scroll state is. Something like this:


`[⏴] [tab] [tab] ][⏵] [➕]` where `➕` is always shown, and never scrolled.

The scroll buttons should probably just be hard-coded here in this function

Reply:
Okay, so just to confirm: You're suggesting we just moved `⏵` and `⏴` into the function, so they are fixed in their behaviour but with `+` exclusively included in `top_bar_right_ui`? Sounds good, it's a better implementation.